tests/unit/test_advanced.py

```diff
@@ -150,22 +161,220 @@ def test_do_attempt_final(self):
                 }
             },
             "finished": True,
-            "num_attempts": 0,
-            'overall_feedback': self.FINAL_FEEDBACK,
+            "attempts": 0,
+            'overall_feedback': [self._make_feedback_message(self.FINAL_FEEDBACK)],
         }
         self.assertEqual(expected_state, self.call_handler('get_user_state', method="GET"))
 
+    def test_do_attempt_not_available(self):
+        """
+        Tests that do_attempt handler returns 400 error for standard mode DnDv2
+        """
+        res = self.call_handler(self.DO_ATTEMPT_HANDLER, expect_json=False)
+
+        self.assertEqual(res.status_code, 400)
 
+
+@ddt.ddt
 class AssessmentModeFixture(BaseDragAndDropAjaxFixture):
     """
     Common tests for drag and drop in assessment mode
     """
-    def test_do_attempt_in_assessment_mode(self):
-        item_id, zone_id = 0, self.ZONE_1
-        data = {"val": item_id, "zone": zone_id, "x_percent": "33%", "y_percent": "11%"}
-        res = self.call_handler('do_attempt', data)
-        # In assessment mode, the do_attempt doesn't return any data.
-        self.assertEqual(res, {})
+    CORRECT_SOLUTION = {}
+
+    @staticmethod
+    def _make_submission(item_id, zone_id):
+        x_percent, y_percent = str(random.randint(0, 100)) + '%', str(random.randint(0, 100)) + '%'
+        data = {"val": item_id, "zone": zone_id, "x_percent": x_percent, "y_percent": y_percent}
+        return data
+
+    def _submit_solution(self, solution):
+        for item_id, zone_id in solution.iteritems():
+            data = self._make_submission(item_id, zone_id)
+            self.call_handler(self.DROP_ITEM_HANDLER, data)
+
+    def _submit_complete_solution(self):
+        self._submit_solution(self.CORRECT_SOLUTION)
+
+    def _submit_partial_solution(self):  # pylint: disable=no-self-use
```

Comment:
@cahrens This class is intended to contain assessment mode tests that does not depend on concrete items/zones configuration. For some tests, we need a partial solution to be submitted, but don't actually care about what items were submitted or not.

That was a motivation. From technical perspective, I want `AssessmentModeFixture` be an abstract class, while keeping as much common tests as possible. In that case, it allows to introduce more data-tied tests easier. So far we've got only one assessment test suite: `TestDragAndDropAssessmentData`, but with `AssessmentModeFixture` it would be easier to add more (similar to `TestDragAndDropHtmlData` - `TestDragAndDropPlainData` - `TestOldDataFormat`).

To make it more "simmetrical", I've made `_submit_correct_solution` raise `NotImplementedError` as well.